### PR TITLE
(CDAP-14646) Use a separate HTTP server for log query

### PR DESCRIPTION
**Pull request by chtyim**
This is to prepare for running the log query inside the same space as the log saver in order to have access to logs on disk.

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/service/LogQueryService.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import org.apache.twill.common.Cancellable;
+import org.apache.twill.discovery.Discoverable;
+import org.apache.twill.discovery.DiscoveryService;
+
+import java.util.Set;
+
+/**
+ * Provides the HTTP server for querying logs
```
Comment (CuriousVini) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/service/LogQueryService.java]: nit: period at the end.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/preview/DefaultPreviewManagerTest.java [hunk 2]
```diff
@@ -92,7 +91,6 @@ public static void beforeClass() throws IOException {
       new NonCustomLocationUnitTestModule(),
       new LocalLogAppenderModule(),
       new LogReaderRuntimeModules().getInMemoryModules(),
```
Comment (CuriousVini) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/preview/DefaultPreviewManagerTest.java:93]: should this include LogQueryServerModule?

**Review by CuriousVini — APPROVED** [review]
just one question. Code changes lgtm 👍

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/preview/DefaultPreviewManagerTest.java (hunk 2 above)
Comment (chtyim) [reply cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/preview/DefaultPreviewManagerTest.java:93]: No. It is not needed for creating DefaultPreviewManager.

**Comment by chtyim** [comment]
Build passed https://builds.cask.co/browse/CDAP-DUT6760

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/service/LogQueryService.java (hunk 1 above)
Comment (chtyim) [reply cdap-watchdog/src/main/java/co/cask/cdap/logging/service/LogQueryService.java]: fixed

**Pull request merged**